src/views/signup.tsx

```diff
@@ -0,0 +1,224 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { useMutation } from "@apollo/client";
+import {
+    OuterContainer,
+    OuterForm,
+    FormInner,
+    Header,
+    Body,
+    Button,
+    SubText,
+    InputEl,
+    SignInLink,
+} from "../styles/CreateAccountElements";
+import { ErrorMessage } from "../styles/ErrorMessageElements";
+import Logo from "../assets/logo-coloured.png";
+import MailLock from "../assets/mail-lock.png";
+import PassLock from "../assets/lock.png";
+import { useNavigate } from "react-router-dom";
+import mutations from "../settings/graphql-mutations";
+
+const SignUpView = () => {
+    // const [data, setData] = useState();
+    let navigate = useNavigate();
+    const [formState, setFormState] = useState(false);
+    const [signUp, signUpResult] = useMutation(mutations.signUp);
+    const [errors, setErrors] = useState({});
+    const [inputs, setInputs] = useState<{ email?: string; password?: string }>(
+        {}
+    );
+
+    const useForm = React.useCallback(
+        (initialValues: any) => {
+            const handleSubmit = (e: any) => {
+                if (e) {
+                    e.preventDefault();
+                }
```

Comment:
Consider `e?.preventDefault();`

Reply:
Hmm, what is the difference between the two?